### PR TITLE
DCOS-46125: DataLayer + Extension-Kid = 🎉

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

- Check if the repositories tab and jobs tab still behave like before

## Trade-offs

- Decided against using external tooling to merge the schemas. Most tooling I found needs to use graphql introspection queries, which we don't have. But we also don't need them, we got all the schema information available to us, so there is no real need. The only "problem" there was is that I can not work on `GraphQLSchema`, making it a little less generic, but again we don't need this. We now touch Schema and resolvers separately, which is also quite nice for testing.

## Dependencies

-

## Video guide

https://drive.google.com/drive/u/0/folders/1MA_XLR8w4HV50fzDDxzaFwYyj-XPg3Sh

### CODE REVIEW

File: src/js/dataLayer.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { graphqlObservable } from "data-service";
+import { makeExecutableSchema, IResolvers } from "graphql-tools";
+import { Observable } from "rxjs/Observable";
+import { injectable, inject, named } from "inversify";
+import { ExtensionProvider } from "extension-kid";
+import { GraphQLSchema } from "graphql";
+import deepmerge from "deepmerge";
```
Comment (DanielMSchmidt) [inline src/js/dataLayer.ts]: This is inside makeExecutableSchema

File: packages/@extension-kid/data-layer/__tests__/dataLayer-test.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+    const dataLayerContainerModule = dataLayerContainerModuleFactory();
+
+    container.load(dataLayerContainerModule);
+  });
+
+  it(
+    "does not error without an extension",
+    marbles(m => {
+      const dl: DataLayer = container.get<DataLayer>(DataLayerType);
```
Comment (TattdCodeMonkey) [inline packages/@extension-kid/data-layer/__tests__/dataLayer-test.ts]: You only need one of `: DataLayer` or `get<DataLayer>`.

File: tsconfig.json [hunk 3]
```diff
@@ -28,6 +28,7 @@
     "skipLibCheck": true,
     "paths": {
       "@dcos/http-service": ["src/typings/dcos__http-service"],
+      "@extension-kid/data-layer": ["packages/@extension-kid/data-layer"],
```
Comment (TattdCodeMonkey) [inline tsconfig.json]: should we add `notification-service` and `toast-notifications` to this config too? We can do that in another PR I just didn't know about this list.

File: plugins/jobs/src/js/data/JobModel.ts [hunk 4]
```diff
@@ -301,17 +303,29 @@ export const resolvers = ({
...
+  pollingInterval: Config.getRefreshRate(),
+  runJob,
+  createJob,
+  updateJob,
+  updateSchedule,
+  deleteJob,
+  stopJobRun
 });
+
+const JobType = Symbol("Job");
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply plugins/jobs/src/js/data/JobModel.ts]: So my thinking was here we create the symbol and we can use it somewhere else by importing Symbol.for or importing this symbol directly

File: plugins/jobs/src/js/components/JobsFormModal.tsx [hunk 5]
```diff
@@ -66,6 +66,7 @@ interface JobFormModalState {
   isConfirmOpen: boolean;
 }
 
+const dataLayer: DataLayer = container.get<DataLayer>(DataLayerType);
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply plugins/jobs/src/js/components/JobsFormModal.tsx]: Hmm, I see your point, I am wondering if we can get completely rid of them by just putting it into the React Context :) 

Would you block the PR because of this or is it more of a nit-pick concern (I rebased this really a lot and we got some new features that could have already used this that now need refactoring, that's why I'm more on the pragmatic side here)

File: packages/@extension-kid/data-layer/__tests__/dataLayer-test.ts (hunk 2 above)
Comment (DanielMSchmidt) [reply packages/@extension-kid/data-layer/__tests__/dataLayer-test.ts]: Okay, I'll remove of one them 💪

File: plugins/jobs/src/js/__tests__/jobsRunNow-test.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ jest.mock("@dcos/data-service", () => ({
   graphqlObservable: jest.fn()
 }));
 
-describe("JobsRunNow", function() {
+describe.skip("JobsRunNow", function() {
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply plugins/jobs/src/js/__tests__/jobsRunNow-test.js]: Sure, thanks for reminding, I must have missed it 😇

File: tsconfig.json (hunk 3 above)
Comment (DanielMSchmidt) [reply tsconfig.json]: If they don't need to be required with types we can leave them of, but we could also add them for the sake of completeness :)

File: plugins/jobs/src/js/data/JobModel.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (TattdCodeMonkey) [reply plugins/jobs/src/js/data/JobModel.ts]: If I'm understanding the docs correctly that won't work. `Symbol.for` registers the Symbol and `Symbol()` doesn't, it returns a unique symbol on every call. So using `Symbol()` we have to always import the same variable. using `Symbol.for()` we could do either.

File: plugins/jobs/src/js/components/JobsFormModal.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (TattdCodeMonkey) [reply plugins/jobs/src/js/components/JobsFormModal.tsx]: I wouldn't block for this as we can improve forward, but I wanted to discuss it. Would also like to get @nLight thoughts on this.

**Comment by DanielMSchmidt** [comment]
@TattdCodeMonkey Your comments have been addressed :)
